jobs/integration/validation.py

```diff
@@ -196,36 +196,42 @@ async def reset_audit_config(control_plane_app, tools):
 
 # START TESTS
 async def test_auth_file_propagation(model, tools):
-    """Validate that changes to /root/cdk/basic_auth.csv on the leader master
-    unit are propagated to the other master units.
+    """Validate that changes to /root/cdk/serviceaccount.key on the leader
+    control-plane unit are propagated to the other control-plane units.
 
     """
     # Get a leader and non-leader unit to test with
     masters = model.applications["kubernetes-control-plane"]
     if len(masters.units) < 2:
-        pytest.skip("Auth file propagation test requires multiple masters")
+        pytest.skip("Test requires multiple control-plane units")
 
     for master in masters.units:
         if await master.is_leader_from_status():
             leader = master
         else:
             follower = master
 
-    # Change basic_auth.csv on the leader, and get its md5sum
+    # Change serviceaccount.key on the leader, and get its md5sum
     leader_md5 = await run_until_success(
         leader,
-        "echo test,test,test >> /root/cdk/basic_auth.csv && "
-        "md5sum /root/cdk/basic_auth.csv",
+        "echo '#EXTRA' >> /root/cdk/serviceaccount.key && "
+        "md5sum /root/cdk/serviceaccount.key",
     )
 
     # Check that md5sum on non-leader matches
-    await run_until_success(
-        follower, 'md5sum /root/cdk/basic_auth.csv | grep "{}"'.format(leader_md5)
+    max_wait = 120
```

Comment:
I'm a little worried that this timeout is too low, given that kubernetes-control-plane hooks usually take longer than a minute to run. A poorly timed update-status hook could prevent a leader-settings-changed hook from writing the new files quick enough. I would be tempted to put this at 5-10 minutes instead.

This is just gut feeling though. I'd be happy to try this at 120 and see what happens.

Reply:
i've got 5 min to burn if you do